src/nodes/mod.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License.  You may obtain a copy
+// of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+
+use fuse;
+use std::ffi::OsStr;
+use std::sync::Arc;
+
+mod dir;
+pub use self::dir::Dir;
+
+/// Type that represents an error understood by the kernel.
+#[derive(Debug, Fail)]
+#[fail(display = "errno={}", errno)]
+pub struct KernelError {
+    errno: i32,
+}
+
+impl KernelError {
+    /// Constructs a new error given a raw errno code.
+    fn from_errno(errno: i32) -> KernelError {
+        KernelError { errno }
+    }
+
+    /// Obtains the errno code contained in this error, which can be fed back into the kernel.
+    pub fn errno(&self) -> i32 {
+        return self.errno;
+    }
+}
```

Comment:
Does it make sense to use [`std::io::Error`](https://doc.rust-lang.org/std/io/struct.Error.html)? It supports going from/to errno (on unix).